fs-storage/src/base_storage.rs

```diff
@@ -0,0 +1,36 @@
+use std::collections::BTreeMap;
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::str::FromStr;
+
+use data_error::Result;
+
+pub trait BaseStorage<K, V>: AsRef<BTreeMap<K, V>>
+where
+    K: FromStr + Hash + Eq + Ord + Debug + Clone,
+    V: Debug + Clone,
+{
+    fn get(&self, id: &K) -> Option<&V>;
```

Comment:
`get` will automatically be provided by `AsRef<Btree..`. An explicit `get` is not necessary.

Reply:
2 min

Reply:
`get` is indeed not necessary but I find it convenient to have. Additionally, if a struct implements `get`, it implies that it also implements `set`

instead of doing
```rust
let data = file_storage.as_ref();
let key_1_value = data.get("key1").unwrap();
```
we can just do
```rust
let key_1_value = file_storage.get("key1").unwrap();
```
What do you think?

Reply:
A get generally implies a set, especially coming from the OOP paradigm. In this case, we are getting a get it is from the BtreeMap which the FileStorage can be treated as. So I don't see a need for an explicit get especially if the get is calling the BtreeMap get again.

Reply:
No problem, it's a detail. 
Let's focus on what matters most. Moving forward with the PR